### ISSUE TITLE
Danish trf wordpiece tokenisations strips accent and lack of lemmatisation list

### ISSUE THREAD

**Issue by KennethEnevoldsen** [explosion/spaCy#8705]
## How to reproduce the behaviour

The Danish transformer strip accent leading to the same wordpieces of meaningfully different words.
```
>>> import spacy
>>> nlp = spacy.load('da_core_news_trf')
>>> doc = nlp("sål og sal er to forskellige ord") # sole and hall is two different words
>>> doc._.trf_data.wordpieces
WordpieceBatch(strings=[['[CLS]', 'sal', 'og', 'sal', 'er', 'to', 'forskellige', 'ord', '[SEP]']], input_ids=array([[   2, 2114,   28, 2114,   33,  385,  599, 1014,    3]]), attention_mask=array([[1, 1, 1, 1, 1, 1, 1, 1, 1]]), lengths=[9], token_type_ids=array([[0, 0, 0, 0, 0, 0, 0, 0, 0]]))
```

The solution is to set strip accents to false and retrain the model. As so:
```
[components.transformer.model.tokenizer_config]
use_fast = true
strip_accents = false
```

I also looked at the lemmatization list used in the Danish model which refers to a directory that contains no lemmatization list for danish. This seems like an error? Regardless the center for Danish language technology (sprogteknologi.dk) has a published [list](https://sprogteknologi.dk/dataset/dictionary-for-the-cst-lemmatizer).

## Info about spaCy
- **spaCy version:** 3.1.0
- **Platform:** macOS-10.16-x86_64-i386-64bit
- **Python version:** 3.8.8
- **Pipelines:** en_core_web_lg (3.0.0), da_core_news_trf (3.1.0), da_core_news_lg (3.0.0), da_core_news_sm (3.0.0), da_core_news_md (3.0.0), en_core_web_md (3.0.0), en_core_web_sm (3.0.0)

**Comment by adrianeboyd** [comment]
Thanks for the notes, we can get that updated for the next version. The lemmatizer history is a bit hard to follow because they were originally in the core library but then moved to the `spacy-lookups-data` repo, so that's my mistake in tracking down the history. I think found the original PR #2184 with the data from this source: https://cst.ku.dk/sto_ordbase/ . Does that look correct?

I'm less sure about what is going on with the transformer tokenizer config. Shouldn't this be specified by the model? Is there a tokenizer config missing somewhere, or a default that changed in `transformers` that's causing this? It looks like `strip_accents` uses the value from `do_lower_case`, which is `True` by default, and this is an uncased model, so ending up with that setting makes sense. Are you sure that the model was trained with accents?

**Comment by KennethEnevoldsen** [comment]
Oh then I think it is the same as the one by sprogteknologi.dk.

Well, accent and casing are two different things (so at least the default here is a bit odd). If you look into the [vocab.txt](https://huggingface.co/Maltehb/danish-bert-botxo/blob/main/vocab.txt) it also seems to have the accents. I agree that it should be a part of loading the model, but it does not seem to be.

**Comment by adrianeboyd** [comment]
I'm just looking at the source to try to figure out what's going on.

https://github.com/huggingface/transformers/blob/90178b0cefe94fef258a39cff5019b5ec150597b/src/transformers/models/bert/tokenization_bert.py#L404-L411

Maybe we should switch to a model where the default config is correct rather than trying to hack this into our config?

The `model_max_length` default that was removed as a default is also a frustrating situation where the model creators didn't really do anything wrong. But here it looks more like something went wrong?

This means it's also not a good recommendation for the quickstart, sigh.

**Comment by KennethEnevoldsen** [comment]
I hear you. 

Currently the altenative transformers for Danish is the Danish Ælæctra cased and uncased (a small ELECTRA). This doesn't strip accents (checked). The model is also trained on the Danish gigawords which is most likely a better corpus the Danish BERT by botxo. Otherwise you will need to look at the Danish transformers.

For conv. we did do the comparison of Danish models here:

https://github.com/centre-for-humanities-computing/DaCy/tree/main/training

To be totally honest I expect a (couple of) new Danish transformers soon.

**Comment by adrianeboyd** [comment]
I don't think either of the `strip_accents` options corresponds to the preprocessing that was done when training the models (strip everything except Danish accents), so I think it might make sense to leave this as it is for now. It's not ideal, but I hope the fine-tuned model will mostly compensate for the mismatch / ambiguity. Let's hope there are better options in the future...
